File: app/src/main/java/ru/melodin/fast/common/ThemeManager.java
package ru.melodin.fast.common;

import android.graphics.Color;

import androidx.annotation.ColorInt;
import androidx.annotation.ColorRes;
import androidx.annotation.IntegerRes;
import androidx.core.content.ContextCompat;

import org.greenrobot.eventbus.EventBus;
import org.jetbrains.annotations.Contract;

import ru.melodin.fast.R;
import ru.melodin.fast.fragment.FragmentSettings;

public class ThemeManager {

    public static final String KEY_THEME_UPDATE = "theme_update";
    private static boolean dark;
    @IntegerRes
    private static int currentTheme, popupTheme, loginTheme, alertTheme;
    @ColorInt
    private static int primary, primaryInverse, primaryDark, accent, main, secondary, background;

    public static void switchTheme(boolean dark) {
        AppGlobal.preferences.edit().putBoolean(FragmentSettings.KEY_DARK_STYLE, dark).apply();
        init();
        EventBus.getDefault().post(new Object[]{KEY_THEME_UPDATE});
    }

    public static void toggleTheme() {
        switchTheme(!isDark());
    }

    static void init() {
        dark = AppGlobal.preferences.getBoolean(FragmentSettings.KEY_DARK_STYLE, false);

        currentTheme = isDark() ? R.style.AppTheme_Dark : R.style.AppTheme_Light;
        popupTheme = isDark() ? R.style.ThemeOverlay_AppCompat : R.style.ThemeOverlay_AppCompat_Light;
        loginTheme = isDark() ? R.style.AppTheme_Login_Dark : R.style.AppTheme_Login_Light;
        alertTheme = isDark() ? R.style.AlertDialog_Theme_Dark : R.style.AlertDialog_Theme_Light;

        primary = getColor(isDark() ? R.color.dark_primary : R.color.primary);
        primaryInverse = getColor(isDark() ? R.color.primary : R.color.dark_primary);
        primaryDark = getColor(isDark() ? R.color.dark_primary_dark : R.color.primary_dark);
        accent = getColor(isDark() ? R.color.dark_accent : R.color.accent);
        background = getColor(isDark() ? R.color.dark_background : R.color.background);
        main = isDark() ? Color.WHITE : Color.BLACK;
        secondary = isDark() ? Color.LTGRAY : Color.DKGRAY;

    }

    @Contract(pure = true)
    public static boolean isDark() {
        return dark;
    }

    @Contract(pure = true)
    public static int getPrimaryInverse() {
        return primaryInverse;
    }

    @Contract(pure = true)
    public static int getCurrentTheme() {
        return currentTheme;
    }

    @Contract(pure = true)
    public static int getLoginTheme() {
        return loginTheme;
    }

    @Contract(pure = true)
    public static int getPopupTheme() {
        return popupTheme;
    }

    @Contract(pure = true)
    public static int getAlertTheme() {
        return alertTheme;
    }

    @Contract(pure = true)
    public static int getPrimary() {
        return primary;
    }

    @Contract(pure = true)
    public static int getPrimaryDark() {
        return primaryDark;
    }

    @Contract(pure = true)
    public static int getAccent() {
        return accent;
    }

    @Contract(pure = true)
    public static int getMain() {
        return main;
    }

    @Contract(pure = true)
    public static int getSecondary() {
        return secondary;
    }

    @Contract(pure = true)
    public static int getBackground() {
        return background;
    }

    @ColorInt
    private static int getColor(@ColorRes int resId) {
        return ContextCompat.getColor(AppGlobal.getContext(), resId);
    }
}

File: app/src/main/java/ru/melodin/fast/common/ThemeManager.kt
package ru.melodin.fast.common

import android.graphics.Color
import androidx.annotation.ColorInt
import androidx.annotation.ColorRes
import androidx.appcompat.app.AppCompatDelegate
import org.greenrobot.eventbus.EventBus
import ru.melodin.fast.R
import ru.melodin.fast.fragment.FragmentSettings
import ru.melodin.fast.util.Keys

object ThemeManager {

    @ColorInt
    private fun getColor(@ColorRes resId: Int): Int {
        return AppGlobal.res.getColor(resId)
    }

    internal fun init() {
        IS_DARK = AppGlobal.preferences.getBoolean(FragmentSettings.KEY_DARK_STYLE, true)

        AppCompatDelegate.setDefaultNightMode(if (IS_DARK) AppCompatDelegate.MODE_NIGHT_YES else AppCompatDelegate.MODE_NIGHT_NO)

        CURRENT_THEME = R.style.AppTheme
        POPUP_THEME = if (IS_DARK) R.style.ThemeOverlay_AppCompat else R.style.ThemeOverlay_AppCompat_Light
        LOGIN_THEME = R.style.AppTheme_Login
        ALERT_THEME = R.style.AlertDialog_Theme

        PRIMARY = getColor(if (IS_DARK) R.color.dark_primary else R.color.primary)
        PRIMARY_INVERSE = getColor(if (IS_DARK) R.color.primary else R.color.dark_primary)
        PRIMARY_DARK = getColor(if (IS_DARK) R.color.dark_primary_dark else R.color.primary_dark)
        ACCENT = getColor(if (IS_DARK) R.color.dark_accent else R.color.accent)
        BACKGROUND = getColor(if (IS_DARK) R.color.dark_background else R.color.background)
        MAIN = if (IS_DARK) Color.WHITE else Color.BLACK
        SECONDARY = if (IS_DARK) Color.LTGRAY else Color.DKGRAY
    }

    fun switchTheme(dark: Boolean) {
        AppGlobal.preferences.edit().putBoolean(FragmentSettings.KEY_DARK_STYLE, dark).apply()
        init()
        EventBus.getDefault().post(arrayOf<Any>(Keys.THEME_UPDATE))
    }

    fun toggleTheme() {
        switchTheme(!IS_DARK)
    }

    var IS_DARK = false

    var CURRENT_THEME = 0
    var POPUP_THEME = 0
    var LOGIN_THEME = 0
    var ALERT_THEME = 0

    var PRIMARY = 0
    var PRIMARY_INVERSE = 0
    var PRIMARY_DARK = 0
    var ACCENT = 0
    var MAIN = 0
    var SECONDARY = 0
    var BACKGROUND = 0

}
